src/parser.c

```diff
@@ -2326,16 +2326,63 @@ basic_block_t *read_body_statement(block_t *parent, basic_block_t *bb)
         lex_expect(T_open_bracket);
 
         /* setup - execute once */
+        basic_block_t *setup = bb_create(parent);
+        bb_connect(bb, setup, NEXT);
+
         if (!lex_accept(T_semicolon)) {
-            lex_peek(T_identifier, token);
-            read_body_assignment(token, parent, OP_generic, &bb);
+            if (!lex_peek(T_identifier, token))
+                error("Unexpected token");
+
+            int find_type_flag = lex_accept(T_struct) ? 2 : 1;
+            type = find_type(token, find_type_flag);
+            if (type) {
+                var = require_var(parent);
```

Comment:
The scope of the declaration in the first clause should be different from the others that outside of the for-statement. i.e. you need to create another scope with `read_code_block()` or something else first.